### PR TITLE
[SYCL][Doc] Update free function queries extension

### PR DESCRIPTION
**Pull request by Pennycook**
- Removes get_linear_id(), get_id() and get_item()
- Renames get_group() to get_work_group()
- Adds deprecated functionality section
- Renames this_kernel namespace to this_work_item

### CODE REVIEW

File: sycl/doc/extensions/proposed/sycl_ext_oneapi_free_function_queries.asciidoc [hunk 1]
```diff
@@ -16,71 +18,71 @@
...
-This extension provides a feature-test macro as described in the core SYCL
-specification section 6.3.3 "Feature test macros". Therefore, an implementation
-supporting this extension must predefine the macro `SYCL_EXT_ONEAPI_FREE_FUNCTION_QUERIES`
-to one of the values defined in the table below. Applications can test for the
-existence of this macro to determine if the implementation supports this
-feature, or applications can test the macro's value to determine which of the
-extension's APIs the implementation supports.
+This extension is written against the SYCL 2020 revision 8 specification.  All
+references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
```
Comment (gmlueck) [inline sycl/doc/extensions/proposed/sycl_ext_oneapi_free_function_queries.asciidoc:43]: There should be a [Status](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/template.asciidoc#status) section here after Dependencies.  Use the suggested text under "Use this if the extension is a proposed new version of an existing extension".

**Review by gmlueck — COMMENTED** [review]
I think this looks good.  Just one comment below.

File: sycl/doc/extensions/proposed/sycl_ext_oneapi_free_function_queries.asciidoc (hunk 1 above)
Comment (Pennycook) [reply sycl/doc/extensions/proposed/sycl_ext_oneapi_free_function_queries.asciidoc:43]: Oops.  Fixed in 49df15411e0e06f9c61c83cc7b91ac0a86655adc.

**Review by gmlueck — APPROVED** [review]

**Comment by Pennycook** [comment]
@intel/llvm-gatekeepers I think this can be merged now.

**Pull request merged**